SixTrack/CMakeLists.txt

```diff
@@ -0,0 +1,1079 @@
+	endif(STATIC)
+
+	find_package(HDF5 COMPONENTS Fortran REQUIRED)
+endif(HDF5)
+
+if(HDF5_FOUND)
+	#uncomment if you want more information
+	message(STATUS "HDF5_INCLUDE_DIRS: ${HDF5_INCLUDE_DIRS}")
+	##message(STATUS "HDF5_INCLUDE_DIR: ${HDF5_INCLUDE_DIR}")
+	message(STATUS "HDF5_DEFINITIONS: ${HDF5_DEFINITIONS}")
+	message(STATUS "HDF5_Fortran_LIBRARIES: ${HDF5_Fortran_LIBRARIES}")
+	##message(STATUS "HDF5_Fortran_HL_LIBRARIES: ${HDF5_Fortran_HL_LIBRARIES}")
+	##message(STATUS "HDF5_HL_LIBRARIES: ${HDF5_HL_LIBRARIES}")
+	##message(STATUS "HDF5_C_LIBRARIES: ${HDF5_C_LIBRARIES}")
+	##message(STATUS "HDF5_CXX_LIBRARIES: ${HDF5_CXX_LIBRARIES}")
+	message(STATUS "HDF5_LIBRARIES: ${HDF5_LIBRARIES}")
+	##message(STATUS "HDF5_FOUND: ${HDF5_FOUND}")
+	##message(STATUS "HDF5_VERSION: ${HDF5_VERSION}")
+	message(STATUS "HDF5_LIBRARY_DIRS: ${HDF5_LIBRARY_DIRS}")
+	##message(STATUS "HDF5_IS_PARALLEL: ${HDF5_IS_PARALLEL}")
+	##message(STATUS "HDF5_Fortran_COMPILER_EXECUTABLE: ${HDF5_Fortran_COMPILER_EXECUTABLE}")
+
+	SET_TARGET_PROPERTIES(SixTrack PROPERTIES COMPILE_FLAGS "-I${HDF5_INCLUDE_DIRS} ${HDF5_DEFINITIONS}")
+endif(HDF5_FOUND)
+
+#Check for OpenMP
+if(OPENMP)
+	find_package(OpenMP REQUIRED)
+endif(OPENMP)
+
+#If we have OpenMP enabled, set the compiler flags needed
+if(OPENMP_FOUND)
+	SET_TARGET_PROPERTIES(SixTrack PROPERTIES COMPILE_FLAGS "${OpenMP_CXX_FLAGS}")
+	SET_TARGET_PROPERTIES(SixTrack PROPERTIES LINK_FLAGS "${OpenMP_CXX_FLAGS}")
+endif()
+
+#Possible future  MPI test
+if(CMAKE_BUILD_TYPE STREQUAL MPI OR CMAKE_BUILD_TYPE STREQUAL MPIDEBUG)
+	FIND_PACKAGE(MPI REQUIRED)
+	if(NOT MPI_CXX_FOUND)
+		MESSAGE(FATAL_ERROR "MPI build requested but no MPI libraries found!")
+	endif()
+	if(MPI_CXX_FOUND)
+		message(STATUS "Using MPI C++ compiler: ${MPI_CXX_COMPILER}")
+		message(STATUS "Using MPI runtime: ${MPIEXEC}")
+		message(STATUS "Using MPI process number switch: ${MPIEXEC_NUMPROC_FLAG}")
+		message(STATUS "Using MPI pre-flags: ${MPIEXEC_PREFLAGS}")
+		message(STATUS "Using MPI post-flags: ${MPIEXEC_POSTFLAGS}")
+		message(STATUS "Using MPI includes: ${MPI_CXX_INCLUDE_PATH}")
+		message(STATUS "Using MPI libraries: ${MPI_CXX_LIBRARIES}")
+		message(STATUS "Using MPI compile flags: ${MPI_CXX_COMPILE_FLAGS}")
+		message(STATUS "Using MPI link flags: ${MPI_CXX_LINK_FLAGS}")
+
+		#We need the MPI headers to be included!
+		include_directories(${MPI_CXX_INCLUDE_PATH})
+	endif()
+endif()
+
+#possible future CUDA test.
+if(CUDA)
+	find_package(CUDA REQUIRED)
+	if(NOT CUDA_FOUND)
+		MESSAGE(FATAL_ERROR "CUDA enabled build requested but CUDA was not found!")
+	endif()
+	if(CUDA_FOUND)
+		message(STATUS "Found CUDA version ${CUDA_VERSION_STRING} at ${CUDA_TOOLKIT_ROOT_DIR}")
+		message(STATUS "Includes at ${CUDA_INCLUDE_DIRS}")
+		ADD_DEFINITIONS("-I${CUDA_INCLUDE_DIRS}")
+	endif()
+endif(CUDA)
+
+#Possible check for Root
+if(ROOT)
+	#/usr/share/root/cmake/FindROOT.cmake
+	SET(CMAKE_MODULE_PATH /usr/share/root/cmake/)
```

Comment:
This does not look very flexible to hard-code specific paths? If so, shouldn't it rather be at least
`set(CMAKE_MODULE_PATH /usr/share/root/cmake/ ${CMAKE_MODULE_PATH})`?

Consider rather adding a folder with the modules that needs specific setup for your case, and modify CMAKE_MODULE_PATH once at the top of file. As an example, you may look [here](https://bitbucket.org/jairhul/bdsim/src/3e88830019953f558c21e86daa04a4f8130aa39d/CMakeLists.txt?at=master&fileviewer=file-view-default), where they have made a folder cmake/Modules containing e.g. FindMadX.cmake and several others.